src/utils/RouteUtils.js

```diff
@@ -91,7 +100,32 @@ async function getRoute(destinationName, end, start, departureTimeQuery, arriveB
     return busRoute;
 }
 
+async function getDetailedRoute(destinationName, end, start, departureTimeQuery, arriveBy) {
+    let { busRoute, walkingRoute } = await fetchBusWalkingRoute(destinationName, end, start, departureTimeQuery, arriveBy);
+    
+    // if there are no bus routes, we should just return walking instead of crashing
+    if (!busRoute && walkingRoute) {
+        return [walkingRoute];
+    }
+
+    // parse the graphhopper bus route
+    busRoute = await ParseRouteUtils.parseDetailedRoute(busRoute, destinationName);
+
+    // combine and filter to create the final route
+    busRoute = await createFinalRoute(
```

Comment:
why not just `return await...` since you never use `busRoute` after it's assigned here?